packages/transformer/src/IModelImporter.ts

```diff
@@ -300,30 +315,22 @@ export class IModelImporter {
       // since default subcategories always exist and always will be inserted after their categories, we treat them as an update
       // to prevent duplicate inserts.
       // Always present elements (0xe, 0x1, 0x10) also will be updated to prevent duplicate inserts.
-      // Otherwise we always insert during a preserveElementIdsForFiltering operation
       if (
         (isSubCategory(elementProps) && isDefaultSubCategory(elementProps)) ||
         this._rootElementIds.has(elementProps.id)
       ) {
         this.onUpdateElement(elementProps);
       } else {
-        // if [[IModelImportOptions.fromTransformation]] is false, try and get the element (update if it exists)
-        const shouldUpdateElement =
-          !this.options.fromTransformation &&
-          this.targetDb.elements.tryGetElement(elementProps.id);
-        if (
-          this._elementsToUpdate.has(elementProps.id) ||
-          shouldUpdateElement
-        ) {
-          tryUpdateElement(elementProps);
-          this._elementsToUpdate.delete(elementProps.id);
+        if (this.elementIdsToUpdateForPreserveId?.has(elementProps.id)) {
```

Comment:
It feels weird to me to have `?.has` . I'm guessing this is necessary because the compiler doesn't know that shouldPreserveElementIdsForFiltering() guarantees that the set is defined?

Maybe the shouldPreserveElementIdsForFiltering function isn't necessary and we should just check the set directly?

Kind of a nitpick I guess

Reply:
I guess you felt the need to do this because we need to check it in onInsertElement as well. Maybe its fine then, feel free to resolve

Reply:
Yeah, the ?. has is there because the compiler can't tell that shouldPreserveElementIdsForFiltering() guarantees that the set is defined